src/Elliptic/DiscontinuousGalerkin/NumericalFluxes/FirstOrderInternalPenalty.hpp

```diff
@@ -0,0 +1,388 @@
+  // of the numerical fluxes, but we wish to impose inhomogeneous (i.e. nonzero)
+  // boundary conditions. Since this contribution does not depend on the
+  // numerical field values, but only on the Dirichlet boundary data, it may be
+  // added as contribution to the source of the elliptic systems. Then, it
+  // remains to solve the homogeneous problem with the modified source.
+  void compute_dirichlet_boundary(
+      const gsl::not_null<db::item_type<::Tags::NormalDotNumericalFlux<
+          FieldTags>>*>... numerical_flux_for_fields,
+      const gsl::not_null<db::item_type<::Tags::NormalDotNumericalFlux<
+          AuxiliaryFieldTags>>*>... numerical_flux_for_auxiliary_fields,
+      const db::item_type<FieldTags>&... dirichlet_fields,
+      const FluxesComputer& fluxes_computer,
+      const db::item_type<FluxesArgs>&... fluxes_args,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& interface_unit_normal)
+      const noexcept {
+    // Need polynomial degress and element size to compute this dynamically
+    const double penalty = penalty_parameter_;
+
+    // Compute n.F_v(u)
+    auto dirichlet_auxiliary_field_fluxes =
+        make_with_value<Variables<tmpl::list<::Tags::Flux<
+            AuxiliaryFieldTags, tmpl::size_t<Dim>, Frame::Inertial>...>>>(
+            interface_unit_normal,
+            std::numeric_limits<double>::signaling_NaN());
+    fluxes_computer.apply(
+        make_not_null(&get<::Tags::Flux<AuxiliaryFieldTags, tmpl::size_t<Dim>,
+                                        Frame::Inertial>>(
+            dirichlet_auxiliary_field_fluxes))...,
+        fluxes_args..., dirichlet_fields...);
+    const auto helper =
+        [&interface_unit_normal, &dirichlet_auxiliary_field_fluxes ](
+            const auto auxiliary_field_tag_v,
+            const auto numerical_flux_for_auxiliary_field) noexcept {
+      using auxiliary_field_tag = std::decay_t<decltype(auxiliary_field_tag_v)>;
+      normal_dot_flux(
+          numerical_flux_for_auxiliary_field, interface_unit_normal,
+          get<::Tags::Flux<auxiliary_field_tag, tmpl::size_t<Dim>,
+                           Frame::Inertial>>(dirichlet_auxiliary_field_fluxes));
+    };
+    EXPAND_PACK_LEFT_TO_RIGHT(
+        helper(AuxiliaryFieldTags{}, numerical_flux_for_auxiliary_fields));
+
+    // Compute 2 * sigma * n.F_u(n.F_v(u))
+    auto principal_dirichlet_auxiliary_field_fluxes =
+        make_with_value<Variables<tmpl::list<
+            ::Tags::Flux<FieldTags, tmpl::size_t<Dim>, Frame::Inertial>...>>>(
+            interface_unit_normal,
+            std::numeric_limits<double>::signaling_NaN());
+    fluxes_computer.apply(
+        make_not_null(
+            &get<::Tags::Flux<FieldTags, tmpl::size_t<Dim>, Frame::Inertial>>(
+                principal_dirichlet_auxiliary_field_fluxes))...,
+        fluxes_args..., *numerical_flux_for_auxiliary_fields...);
+    const auto helper2 = [
```

Comment:
All the helpers are confusingly named. Can you give them individual names describing what they do, like compute_dirichlet_penalty perhaps? Here and throughout, wherever "helper" is used as a name